datacatalog/cloud-client/src/main/java/com/example/datacatalog/CreateCustomType.java

```diff
@@ -0,0 +1,216 @@
+      dataCatalogClient.deleteEntry(entryName);
+      System.out.printf("\nDeleted Entry: %s", entryName);
+    } catch (PermissionDeniedException | NotFoundException e) {
+      // PermissionDeniedException or NotFoundException are thrown if
+      // Entry does not exist.
+      System.out.println("Entry does not exist.");
+    }
+
+    // Delete any pre-existing Entry Group with the same name
+    // that will be used in step 2.
+    try {
+      String entryGroupName = EntryGroupName.of(projectId, location, entryGroupId).toString();
+      dataCatalogClient.deleteEntryGroup(entryGroupName);
+      System.out.printf("\nDeleted Entry Group: %s", entryGroupName);
+    } catch (PermissionDeniedException | NotFoundException e) {
+      // PermissionDeniedException or NotFoundException are thrown if
+      // Entry Group does not exist.
+      System.out.println("Entry Group does not exist.");
+    }
+
+    String tagTemplateName =
+        TagTemplateName.newBuilder()
+            .setProject(projectId)
+            .setLocation(location)
+            .setTagTemplate(tagTemplateId)
+            .build()
+            .toString();
+
+    // Delete any pre-existing Template with the same name
+    // that will be used in step 4.
+    try {
+      dataCatalogClient.deleteTagTemplate(
+          DeleteTagTemplateRequest.newBuilder()
+              .setName(tagTemplateName)
+              .setForce(true)
+              .build());
+      System.out.printf("\nDeleted template: %s", tagTemplateName);
+    } catch (Exception e) {
+      System.out.printf("\nCannot delete template: %s", tagTemplateName);
+    }
+
+    // 2. Create an Entry Group.
+    // Construct the EntryGroup for the EntryGroup request.
+    EntryGroup entryGroup =
+        EntryGroup.newBuilder()
+            .setDisplayName("My awesome Entry Group")
+            .setDescription("This Entry Group represents an external system")
+            .build();
+
+    // Construct the EntryGroup request to be sent by the client.
+    CreateEntryGroupRequest entryGroupRequest =
+        CreateEntryGroupRequest.newBuilder()
+            .setParent(LocationName.of(projectId, location).toString())
+            .setEntryGroupId(entryGroupId)
+            .setEntryGroup(entryGroup)
+            .build();
+
+    // Use the client to send the API request.
+    EntryGroup createdEntryGroup = dataCatalogClient.createEntryGroup(entryGroupRequest);
+
+    System.out.printf("\nEntry Group created with name: %s", createdEntryGroup.getName());
+
+    // 3. Create an Entry.
+    // Construct the Entry for the Entry request.
+    Entry entry =
+        Entry.newBuilder()
+            .setUserSpecifiedSystem("onprem_data_system")
+            .setUserSpecifiedType("onprem_data_asset")
+            .setDisplayName("My awesome data asset")
+            .setDescription("This data asset is managed by an external system.")
+            .setLinkedResource("//my-onprem-server.com/dataAssets/my-awesome-data-asset")
```

Comment:
In general, we have a preference for samples to use UUID's, possibly in conjunction w/ text.  We often have multiple tests running at the same time.